include/dali/core/span.h

```diff
@@ -58,10 +59,30 @@ class span {
   >>
   DALI_HOST_DEV constexpr span(const span<U, Extent>& s) noexcept : data_(s.data()) {}
 
+
   DALI_HOST_DEV constexpr span(pointer firstElem, pointer lastElem) : data_(firstElem) {
     /* assert(lastElem - firstElem == Extent); */
   }
 
+  /**
+   * @remarks Container is not a specialization of span,
+   *          specialization of std::array or an array type
+   */
+  template <typename Container,
+          typename std::enable_if<is_container<Container>::value, int>::type = 0>
+  DALI_HOST_DEV constexpr span(Container &container)
+          : span(container.data(), container.size()) {}
```

Comment:
...see here - `is_container` will accept e.g. `std::list` which will fail to compile here - it shouldn't make it to this point.

Reply:
How is it worse than `make_span`, which will also accept `std::list` and fail to compile? The sole reason, that it's non-explicit isn't, in my opinion, any reason for this being worse.